libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -173,7 +175,8 @@ void DataFlowAnalyzer::operator()(FunctionDefinition& _fun)
 	for (auto const& var: _fun.returnVariables)
 	{
 		m_variableScopes.back().variables.emplace(var.name);
-		handleAssignment({var.name}, nullptr, true);
+		if (m_assumeReturnVariablesZeroInitialized)
```

Comment:
I don't think we should change the knowledeg but the replacement instead.

Reply:
I'd disagree, especially since these steps are the most expensive ones we have in the optimizer already and the alternative is significantly more costly.

Reply:
But well, maybe it's not *that* expensive in the end, so yeah, I'm fine with moving towards #11246.